src/main/java/com/develhope/spring/users/services/UserService.java

```diff
@@ -3,22 +3,51 @@
 import com.develhope.spring.users.components.UserMapper;
 import com.develhope.spring.users.dtos.UserCreatorDTO;
 import com.develhope.spring.users.dtos.UserSavedDTO;
+import com.develhope.spring.users.models.Roles;
 import com.develhope.spring.users.models.User;
 import com.develhope.spring.users.repositories.UserRepository;
+import com.develhope.spring.users.responseStatus.UserNotFoundException;
+import com.develhope.spring.vehicles.dtos.VehicleCreatorDTO;
+import com.develhope.spring.vehicles.models.Vehicle;
+import com.develhope.spring.vehicles.responseStatus.NotAuthorizedOperationException;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import java.util.Optional;
+
 @Service
 public class UserService {
     @Autowired
     private UserRepository userRepository;
     @Autowired
     private UserMapper userMapper;
 
+    private void checkUserAuthorizationBy(long userId) {
+        Optional<User> optionalUser = userRepository.findById(userId);
+        if (optionalUser.isEmpty() || !optionalUser.get().getRoles().equals(Roles.ADMIN)) {
+            throw new NotAuthorizedOperationException("Permesso negato. Non autorizzato ad aggiornare gli utenti");
+        }
+    }
+    private void checkIfUserExists(long userId){
```

Comment:
QUesto non serve. Come diceva oggi lorenzo in live la repository gia lancia l'eccezione per noi. Basta solo catturarla con l exception handler nel controller